site/documentation/content/api/Tenant-API.md

```diff
@@ -153,17 +153,37 @@ The JSON structure below contains example information for tenant `TEST_TENANT`.
 
 ### Trusted CA Format
 
-The table below provides an overview of the members defined for the *trusted-ca* JSON object:
+If the *trusted-ca* property is provided, then it MUST contain a non-empty array of JSON objects, were each JSON object represents a single trusted CA. The table below provides an overview of the members of a JSON object representing a trusted CA:
 
 | Name                     | Mandatory  | Type          | Default Value | Description |
 | :------------------------| :--------: | :------------ | :------------ | :---------- |
-| *subject-dn*             | *yes*      | *string*      |               | The subject DN of the trusted root certificate in the format defined by [RFC 2253](https://www.ietf.org/rfc/rfc2253.txt). |
-| *cert*                   | *no*       | *string*      |               | The Base64 encoded binary DER encoding of the trusted root X.509 certificate. |
-| *public-key*             | *no*       | *string*      |               | The Base64 encoded binary DER encoding of the trusted root certificate's public key. |
+| *subject-dn*             | *yes*      | *string*      |               | The subject DN of the trusted root certificate in the format defined by [RFC 2253](https://www.ietf.org/rfc/rfc2253.txt). Trusted CAs of the same tenant MAY share the same subject DN (e.g. allowing for the definition of overlapping validity periods). However, trusted CAs of different tenants MUST NOT share the same subject DN in order to allow for the look up of a tenant by the subject DN of one of its trusted CAs. |
+| *public-key*             | *yes*       | *string*     |               | The Base64 encoded binary DER encoding of the trusted root certificate's public key. |
+| *not-before*             | *no*       | *string*      |               | The property indicating that the trusted root X.509 certificate is not valid before the given date. The value MUST be an **ISO 8601 compliant** [*combined date and time representation*](https://en.wikipedia.org/wiki/ISO_8601#Combined_date_and_time_representations) in extended format. |
```

Comment:
I do not think that it makes much sense to include trust anchors in the list that are not valid at this particular moment, right? IMHO the service implementation should only return trust anchors that an adapter can use to verify the client cert's signature without the need to filter out those that are *not yet* or *no longer* valid. WDYT?

Reply:
sure 👍 

i think it should be enough if we document this in the API docs of `TenantObject#getTrustAnchors()` method. should we include that in the tenant API spec?

Reply:
FMPOV this *has* to be reflected particularly in the Tenant API. In fact, it means that the trust anchor objects do not (need to) contain a *not-before* nor a *not-after* property because every trust anchor included is *per definitionem* valid and usable for the validation.

Reply:
hmm.. I am confused here..  let me break it down with my little understanding :-)

During tenant creation, all trust anchors (MUST/MAY/COULD)? be valid, but that might not necessary be the case in the future

Lets assume two trust actors (`A` and `B`) are added for a tenant (during tenant creation). trust anchor `A` will expire in 1 year while `B` will expire in 5 years (relative to the time instant in which the tenant was created). From the point the tenant is created and before `A` expires, a call to retrieve the trust anchors for the tenant should return `two valid` trust anchors.

Now lets assume that trust anchor `A` expires but it is still registered with the tenant. Now a call to retrieve the valid trust anchors for the tenant should return a single trust anchor. I think this is where we might need the not-before/not-after properties to perform the validity check?

Reply:
Yes, but I would like the filtering of the relevant trust anchors to be done in the Tenant service implementation, not in the protocol adapter. Therefore, we need to have these properties in the HTTP based management API for tenants, but we do not need it in the payload that is sent to a client of the Tenant API.

Reply:
> Therefore, we need to have these properties in the HTTP based management API for tenants, but we do not need it in the payload that is sent to a client of the Tenant API.

I see now, thanks 👍  I think this distinction is very important. what I find confusing sometimes is the usage of the `TenantObject` to encapsulate information for a client sending a tenant add/update request. I think we should really restrict its usage only for the get operation and remove its occurrence in all tests cases (both in unit and integration tests) that have to do with sending tenant requests

Reply:
that is my opinion as well and that is why I created #1360

Reply:
@sophokles73 I filed #1375. once that is merged, then I will revert revision 91cf13d and continue working on this PR